dns/delegation_set.tf

```diff
@@ -0,0 +1,17 @@
+# ------------------------------------------------------------------------------
+# Create the primary reusable delegation set.
+# It is imperative that this delgation set never be deleted.
+# ------------------------------------------------------------------------------
+
+resource "aws_route53_delegation_set" "main" {
+  # We can't perform this action until our policy is in place.
+  depends_on = [
+    aws_iam_role_policy_attachment.provisionroute53_policy_attachment,
+    aws_iam_policy.provisionroute53_policy
```

Comment:
Did you test if it needs to depend on both the policy and the policy attachment?  Doesn't the policy attachment automatically depend on the policy?

Reply:
I ran it originally with the single attachment dependency.  @jsf9k said that he thinks if the policy is changed this will stop working.  Which makes sense.  So I added a second link to the policy itself.  This doesn't break anything.

Reply:
That works for me.